### DISCUSSION TITLE
Supertakes

### DISCUSSION THREAD

**Discussion by MonkeyBars3k** [MonkeyBars3k/ReaScripts#203] [Ideas]
Idea from @AZPercussion

Is it possible to move Superitem functionality to takes? Are there any logical contradictions? How could this be done?

**Comment by MonkeyBars3k** [comment]
- Takes can hold item data
- Multiple Supertakes could live on the same item with totally unrelated contained items in each, though they would share the same item properties
- Would we have to remove the concept of Superitem and switch to only Supertakes?

**Comment by MonkeyBars3k** [comment]
My main question for now is: how would pool updates propagate on inactive Sibling Supertakes?

This also affects item property propagation but that comes later...

**Reply by AZpercussion** [reply]
I've thought about it.
Definitely, there's no logical ability to propagate SuperItem bounds to inactive takes.

Here are the same limitations as with take system at all.

So, there's no sense for user to put in takes strongly different superitems.
But it could be useful when we need to choose the suitable variant from similar sounds.

**Comment by MonkeyBars3k** [comment]
Yes, Supertakes wouldn't have any length or position propagation, only possibly offset. I'm not really clear on the ideal workflow for Supertakes, so I'm more wondering about a desired UX – and then I could work backwards on implementation.

**Comment by AZpercussion** [comment]
Potential use scenario:

- Create SuperItem from items,
- Duplicate active take with unpooling.

Now we got two unrelated Supertakes in one item.

- Unglue and edit second SuperTake,
- Reglue it,
- Duplicate active take with unpooling.

Now we got three unrelated Supertakes in one item, each of which can be unglued.
And so on.

- Then copy SuperItem through a project
- Choose more suitable SuperTake in each item.
- Edit some takes and their siblings in other items will be updated.
- If some of siblings are active takes, they could use full propagation stuff as current is.

**Reply by Infrabass** [reply]
Hi guys, so as MonkeyBars3k proposed on the forum thread, here is a video detailing a potential use case for SUPERTAKE:
https://dropover.cloud/03968b

To briefly resume it, for sound design (that constantly need variations especially in game audio) it's super handy to be able to implode multiple takes (variations) into the same item. That way we have all the variations of a composite file in a single item that can easily move, duplicate, transpose, etc.

Sometimes (often) we need to delete a part of a take (and therefore an item) and crossfade the 2 parts together to remove an element we don't want (an intempestive bird call for exemple).

Supertake would be super handy there, because we could then treat the now splitted take as a whole again (and so adding take FX, take envelope, etc).

Now it's impossible because Superglue create a new item and therefore remove the ability to switch between takes.

**Comment by MonkeyBars3k** [comment]
If I'm not mistaken, you could achieve this now by changing takes on the contained items. Why do you also need Supertakes on the Superitem to achieve this aim? Are you just trying to save steps? Or perhaps just to keep your project cleaner? 

Another way to achieve this would be to have different versions of the combinations of takes in your Superitem source but outside the bounds of the item itself, although that would require lining them up again to change "versions".

**Reply by Infrabass** [reply]
Sorry my video being without any audio it's maybe not easy to follow.
As I was talking in my third paragraph above, the use case for supertake here will be to be able to glue again takes that have been splitted and crossfaded together (see in the video) while supporting multi-takes item to keep the possibility to switch between takes (variations).
While I'm writing this I realize that it's maybe not very efficient because we would still need to super glue every imploded takes of an item as the split/crossfade affect every imploded takes of the item.
Mmh.

BUT another situation would be this one:
https://dropover.cloud/8b6765

Having 2 superitems that would imploded into one item (and therefore becoming takes). If you want to edit one of these superitems we have a message that say it's not supported. And to be honest I get why because it become complicated as you will have to create a temp track for edition or something like that I guess.

**Reply by MonkeyBars3k** [reply]
> the use case for supertake here will be to be able to glue again takes that have been splitted and crossfaded together (see in the video) while supporting multi-takes item to keep the possibility to switch between takes (variations). While I'm writing this I realize that it's maybe not very efficient because we would still need to super glue every imploded takes of an item as the split/crossfade affect every imploded takes of the item. Mmh.

Yes it gets complicated fast, which is why I created this Discussion. But "to glue again takes that have been splitted and crossfaded together" can be done with takes on contained items, right?? Perhaps I'm not quite getting what you're doing here. Are these the LKC sound design tools I'm looking at?

> And to be honest I get why because it become complicated as you will have to create a temp track for edition or something like that I guess.

Ah yes, in fact AZ has suggested a temp track for editing, and I think it's a great idea. It's a bit lower priority for now though. I'm still fixing bugs in basic actions!

Hey, what is that awesome script that shows the item source up top?? It's amazing to see different Reaper themes and scripts like this . . . like a different program!

**Comment by Infrabass** [comment]
> But "to glue again takes that have been splitted and crossfaded together" can be done with takes on contained items, right?? Perhaps I'm not quite getting what you're doing here
You're right but I constantly try new takes (variations) when sound designing so it would not be very convenient to have to unglue and re-glue each time.
> Are these the LKC sound design tools I'm looking at?
> Hey, what is that awesome script that shows the item source up top?? It's amazing to see different Reaper themes and scripts like this . . . like a different program!
I think this is a mix of nvk awesome scripts mostly dedicated to game audio and mine (that aren't public yet)

Thanks for your dedication to make the most complete new feature in Reaper!


Vincent Fliniaux
Bruits.Studio Co-Founder
Lead Sound Designer & Composer
+1 (438) 829-2410
***@***.***
https://bruits.studio
